src/components/DocumentationTopic/DocumentationNav/VersionToggle.vue

```diff
@@ -0,0 +1,283 @@
+      versionModel: null,
+    };
+  },
+  mounted() {
+    // on resize, re-calculate the width of the select.
+    const cb = debounce(async () => {
+      // we wait for 3 frames, as that is the minimum it takes
+      // for the browser orientation-change transitions to finish
+      await waitFrames(3);
+      this.calculateSelectWidth();
+    }, 150, true);
+    window.addEventListener('resize', cb);
+    window.addEventListener('orientationchange', cb);
+    this.$once('hook:beforeDestroy', () => {
+      window.removeEventListener('resize', cb);
+      window.removeEventListener('orientationchange', cb);
+    });
+  },
+  updated() {
+    this.versionModel = this.currentVersion;
+  },
+  watch: {
+    // interfaceversion: {
+    //   immediate: true,
+    //   handler(version) {
+    //     this.versionModel = version;
+    //   },
+    // },
+    versionModel: {
+      immediate: true,
+      handler: 'calculateSelectWidth',
+    },
+  },
+  methods: {
+    /**
+     * Returns a formatted route object
+     * @param {{ query: string, path: string }} route - a config object passed by the render JSON
+     * @returns {{ path: (null|string), query: { version: (string|undefined) }}}
+     */
+    getRoute(route) {
+      // pass undefined to remove the query param if its most recent version
+      const version = route.query === this.versionList[0] ? undefined : route.query;
+      return {
+        // make sure we dont loose any extra query params on the way
+        query: { ...this.$route.query, version },
+        path: this.isCurrentPath(route.path) ? null : this.normalizePath(route.path),
+      };
+    },
+    // foo(version) {
+    //   this.store.setPreferredVersion(version);
+    // },
+    pushRoute(route) {
+      // Persist the selected version as a preference in the store (backed by
+      // the browser's local storage so that it can be retrieved later for
+      // subsequent navigation without the query parameter present)
+      this.store.setPreferredVersion(route.query);
+      // Navigate to the version variant page
+      this.$router.push(this.getRoute(route));
+    },
+    isCurrentPath(path) {
+      // the `.replace` call is needed since paths vended by the backend do not
+      // include a leading slash, while the router provided path does
+      return this.$route.path.replace(/^\//, '') === path;
+    },
+    normalizePath(path) {
+      // Sometimes `paths` data from `variants` are prefixed with a leading
+      // slash and sometimes they aren't
+      return path.startsWith('/') ? path : `/${path}`;
+    },
+    /**
+     * Calculated the width of the select by fetching it from the faux select.
+     * @return {Promise<void>}
+     */
+    async calculateSelectWidth() {
+      // await next tick, so we are sure the element is rendered.
+      await this.$nextTick();
+      this.adjustedWidth = this.$refs['version-sizer'].clientWidth + 6;
+    },
+  },
+  computed: {
+    singleVersionPage() {
```

Comment:
I don't like the fact that singleVersionPage doesn't return an boolean but you use it like if it would. I would change the name of this computed property to `hasSingleVersion` where I would return a boolean value and then I would directly print in line 47 the first value

Reply:
```js
hasSingleVersion: ({ versionList }) => versionList.length === 1,
```